lib/rules/no-unnecessary-act.ts

```diff
@@ -89,7 +124,16 @@ export default createTestingLibraryRule<[], MessageIds>({
           node: identifierNode,
           messageId: 'noUnnecessaryActEmptyFunction',
         });
-      } else if (!hasSomeNonTestingLibraryCall(blockStatementNode.body)) {
+        return;
+      }
+
+      const isReported =
+        (!isStrict && !hasSomeNonTestingLibraryCall(blockStatementNode.body)) ||
+        (isStrict &&
+          hasSomeNonTestingLibraryCall(blockStatementNode.body) &&
+          hasTestingLibraryCall(blockStatementNode.body));
```

Comment:
I guess this can be simplified like this? What do you think?

```suggestion
const isReported = isStrict
  ? hasSomeNonTestingLibraryCall(blockStatementNode.body) &&
    hasTestingLibraryCall(blockStatementNode.body)
  : !hasSomeNonTestingLibraryCall(blockStatementNode.body);
```